### ISSUE TITLE
Unused annotation for methods implementing an interface and not throwing exception declared on interface

### ISSUE THREAD

**Issue by ondrejbouda** [pepakriz/phpstan-exception-rules#89]
Let's have an interface:
```php
interface UserRepository
{
    /**
     * @throws UserNotFound
     */
    public function get(EmailAddress $emailAddress) : User;
}
```

And an implementation:
```php
final class FakeImplementation implements UserRepository
{
    public function get(EmailAddress $emailAddress) : User
    {
        // return some dummy User, do not throw any exceptions
    }
}
```

`UserNotFound` is a checked exception.

But now I get an error:
```
FakeImplementation.php
Unused @throws UserNotFound annotation
```

I don't think this should be reported for classes implementing an interface. The declaration of thrown exceptions on an interface should mean (in my opinion) that
* the implementation can (but doesn't *need to*) throw these exceptions
* and the implementation cannot throw other exceptions than declared.

I understand that this behaviour makes sense for simple classes with no inheritance. But not for implenetations of interfaces. Is there something I am missing? Is there a way around this?

**Comment by pepakriz** [comment]
Hi. This is expected behavior since phpstan v0.11.0 (PR phpstan/phpstan#1533). If you omit phpdoc, it is inherited from the parent method. So your code is interpreted in the same way like:

```php
interface UserRepository
{
    /**
     * @throws UserNotFound
     */
    public function get(EmailAddress $emailAddress) : User;
}

final class FakeImplementation implements UserRepository
{
    /**
     * @throws UserNotFound
     */
    public function get(EmailAddress $emailAddress) : User
    {
        // return some dummy User, do not throw any exceptions
    }
}
```

But since 0.11.6 (https://github.com/phpstan/phpstan/releases/tag/0.11.6) you can write:
```php
    /**
     * @throws void
     */
    public function get(EmailAddress $emailAddress) : User
    {
        // return some dummy User, do not throw any exceptions
    }
```

**Comment by ondrejbouda** [comment]
Yes, I understand it is intentional. I just don't think it is right behavior in this case. That a method doesn't *need to* throw exception when annotated with `@throws`. It is very limiting when implementing an interface.

But thanks for the `@throws void` tip, this way I can work around this issue!
